### PR TITLE
add deserialization and transform request image file 

### LINKED ISSUE #16: Deserialize model object and give predictions
Once #2 is done, the main API should deserialize the model object, transform the user's image into a numpy array, pass the array to the model to get a prediction, and return a label and confidence % in the response.

```json
{
    "predicted-label": "forged",
    "prediction-confidence-percentage": "85"
}
```

This is assuming the model _does_ give  a confidence for its prediction. If not then leave that field out.

### PR DESCRIPTION
**Pull request by YiminArava4508**
the API should now deserialize the model object and transform the user's image file to a NumPy array for the ml model to predict a response

### CODE REVIEW

File: backend/image-upload-api/flask-api/main.py [hunk 1]
```diff
@@ -1,42 +1,78 @@
...
 
 
-# serving index.html on flask
+# Deserialize the json file and create the model-----------------------------------------------------------------------
+with open("./notebooks/models/2021-09-06-keras.json", "r") as json_file:
+    json_model = json.load(json_file)
+
+model_test = model_from_json(json_model)
+
+# Some functions for resp ---------------------------------------------------------------------------------------------
```
Comment (JasonFreeberg) [inline backend/image-upload-api/flask-api/main.py]: @YiminArava4508 There's a thing called "docstrings" when you put your comments and notes for a method into the method itself, check out this doc: https://www.python.org/dev/peps/pep-0257/

So for this one, it would look like this:

```python
def result(array):
    """
    Given the prediction array from the model, this method returns the predicted label.
    """
    if array[0][0] > array[0][1]:
        return 'genuine'
    elif array[0][0] < array[0][1]:
        return 'forged'
    else:
        return 'unable to determine'
```

File: backend/image-upload-api/flask-api/main.py [hunk 2]
```diff
@@ -1,42 +1,78 @@
...
+def result(array):
+    if array[0][0] > array[0][1]:
+        return 'genuine'
+    elif array[0][0] < array[0][1]:
+        return 'forged'
+    else:
+        return 'unable to determine'
+def result_percentage(array):
```
Comment (JasonFreeberg) [inline backend/image-upload-api/flask-api/main.py:35]: Add a docstring here too and make sure there's a blank line in between the methods

File: backend/image-upload-api/flask-api/main.py [hunk 3]
```diff
@@ -1,42 +1,78 @@
-
+# Imports ------------------------------------------------------------------------------------------------------------
```
Comment (JasonFreeberg) [inline backend/image-upload-api/flask-api/main.py]: We can remove this

File: backend/image-upload-api/flask-api/main.py [hunk 4]
```diff
@@ -1,42 +1,78 @@
...
 from app import app
 from flask import Flask, request, redirect, jsonify, send_from_directory
+import tensorflow as tf
```
Comment (JasonFreeberg) [inline backend/image-upload-api/flask-api/main.py]: Remove this, the import isn't used

File: backend/image-upload-api/flask-api/main.py [hunk 5]
```diff
@@ -1,42 +1,78 @@
-
+# Imports ------------------------------------------------------------------------------------------------------------
 from app import app
 from flask import Flask, request, redirect, jsonify, send_from_directory
```
Comment (JasonFreeberg) [inline backend/image-upload-api/flask-api/main.py]: You can remove the "redirect" import, it's not used anywhere

File: backend/image-upload-api/flask-api/main.py [hunk 6]
```diff
@@ -1,42 +1,78 @@
...
+import tensorflow as tf
+from tensorflow import keras
+import numpy as np
+import json
+from PIL import Image,  ImageOps
```
Comment (JasonFreeberg) [inline backend/image-upload-api/flask-api/main.py]: Tensorflow, keras, numpy, and PIL need to be added to the `requirements.txt` file. You can update the requirements.txt file using `pip freeze > requirements.txt`

**Review by JasonFreeberg — CHANGES_REQUESTED** [review]

**Comment by YiminArava4508** [comment]
I removed some unnecessary code based on the pylint CI, the code is much cleaner now. The rest of the checks are quite weird I am not sure if I should follow the suggestions
Capture

**Comment by JasonFreeberg** [comment]
Yeah don't worry about those last ones, there's something wrong with the CI or how the project files are organized. I'll fix that.

I'll review it tomorrow and merge it in

**Review by JasonFreeberg — APPROVED** [review]
